client/src/components/SignUp/index.js

```
@@ -0,0 +1,162 @@
+import Grid from "@material-ui/core/Grid";
+import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import Container from "@material-ui/core/Container";
+import axios from 'axios';
+import {SERVER_URL} from "../../../config";
+import {Redirect} from 'react-router'
+
+const useStyles = makeStyles(theme => ({
+  "@global": {
+    body: {
+      backgroundColor: theme.palette.common.white
+    }
+  },
+  paper: {
+    marginTop: theme.spacing(8),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center"
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main
+  },
+  form: {
+    width: "100%", // Fix IE 11 issue.
+    marginTop: theme.spacing(3)
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2)
+  },
+  errorMessage : {
+    "text-align" : "center",
+    color : "red",
+  }
+}));
+
+export default function SignUp() {
+  const classes = useStyles();
+
+  const [user,setUser] = useState ({name :"", email:"",password: ""});
+  const [errorMessage, setErrorMessage] = useState('');
+  const [signUpSuccess, setSignUpSucess] = useState(false);
+
+  const handleEmailChange = evt => {
+    setUser({...user,email :evt.target.value});
+  };
+
+  const handlePasswordChange = evt => {
+    setUser({...user, password:evt.target.value});
+  };
+
+  const handleNameChange = evt => {
+    setUser({...user, name : evt.target.value});
+  };
+
+  const postData = async (e) => {
+    console.log(user);
+    e.preventDefault();
+    try {
+    const response = await axios({
+      method : 'post',
+      url : `${SERVER_URL}/signup`,
+      data : {...user},
+      headers: { "Content-Type": "application/json" },
+    });
+    console.log(response);
+    if(response && response.data && response.data.isSuccess) {
+      setErrorMessage('');
+      setSignUpSucess(true);
+    }
```

If it didn't get `response` or `response.data`
... you need to throw an error

like this,
```js
try {

...

if(!response || !response.data) throw new Error('No response from server');
if(!response.data.isSuccess) throw new Error(response.data.message);

// No error, do other things
...

} catch (err) {
    setErrorMessage(err);
}
```

---

Will refactor once snack bar is merged .